### PR TITLE
introduce react-user-tour typings

### PR DESCRIPTION
**Pull request by ccancellieri**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.
- [x] write some good test (please help!)

### CODE REVIEW

File: react-user-tour/react-user-tour.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Type definitions for react-user-tour
+// Project: https://github.com/socialtables/react-user-tour
+// Definitions by: Carlo Cancellieri <https://github.com/ccancellieri>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module "react-user-tour" {
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: You need to add a reference to the `react.d.ts` to import.
Also, use 4 spaces instead of 2 for consistence

**Comment by ccancellieri** [comment]
Job done, should be all right, thanks

File: react-user-tour/react-user-tour-tests.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+/// <reference path="../react/react.d.ts" />
+/// <reference path="../react/react-dom.d.ts" />
+
+import React = require("react");
+import ReactDOM = require("react-dom");
+
+import ReactUserTour from "react-user-tour";
+
+interface State {
+  tourStep:number
```
Comment (zhengbli) [inline react-user-tour/react-user-tour-tests.tsx]: Indentation is not consistent here

File: react-user-tour/react-user-tour-tests.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  isTourActive:boolean
+}
+
+class TestApp extends React.Component<{}, State> {
+
+    constructor(p:any){
+        super(p)
```
Comment (zhengbli) [inline react-user-tour/react-user-tour-tests.tsx]: semicolon at the end

File: react-user-tour/react-user-tour-tests.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        this.setState({
+            isTourActive:true,
+            tourStep:1
+        })
```
Comment (zhengbli) [inline react-user-tour/react-user-tour-tests.tsx]: `;`

File: react-user-tour/react-user-tour-tests.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+                {
+                  step: 1,
+                  selector: ".MyClass",
+                  title: <div>React User Tour</div>,
+                  body: <div>Provide a simple guided tour around a website utilizing css selectors.</div>,
+                  position: "bottom"
+                }
+            ]}
+            />
+
```
Comment (zhengbli) [inline react-user-tour/react-user-tour-tests.tsx]: Are these empty lines necessary?

File: react-user-tour/react-user-tour.d.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+    // Import React
+    import { HTMLAttributes, ComponentClass } from 'react';
+
+    interface TourStep {
+        /**
+        Index number of the step in the tour
+        */
+        step: number
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: `;`  at the end

File: react-user-tour/react-user-tour.d.ts [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+        */
+        title: string
+        /**
+        a react element representing the main body message of the tour step
+        */
+        body: string
+        /**
+        Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
+        */
+        horizontalOffset?:number
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: space between `:` and `number`

File: react-user-tour/react-user-tour.d.ts [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+        /**
+        Optional properties horizontalOffset and verticalOffset values allow to move tooltip around pointed element.
+        */
+        verticalOffset?:number
+        /**
+        Each step can also take an optional argument, position which will override the position of the tour component in relation to the selector that is determined by the application.
+        */
+        position?: 'left'|'right'|'top'|'topLeft'|'bottom'|'bottomLeft'
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: Use `'` or `"` consistently in the file

File: react-user-tour/react-user-tour.d.ts [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+        /**
+        An integer representing the current active step of the tour
+        */
+        step:number
+        /**
+        function that fires when user clicks the Next button.
+        Receives the next step integer as a callback.
+        For example, if current step is 1 and user clicks the Next button, onNext(2) will be called.
+        */
+        onNext:any
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: Why `any`? The description seems to be saying it is a function type

File: react-user-tour/react-user-tour.d.ts [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+        */
+        buttonContainerStyle?:any
+        /**
+        We provide an arrow that points to the selector, but you may optionally pass in your own React element in the place of the arrow provided.
+        */
+        arrow?:any
+        /**
+        If you choose to use the provided arrow, you can set the pixel size here with an integer value.
+        */
+        arrowSize?:any
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: This looks like `number`

File: react-user-tour/react-user-tour.d.ts [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+        /**
+        If you choose to use the provided arrow, you can set the color here by passing in a hex value.
+        */
+        arrowColor?:any
```
Comment (zhengbli) [inline react-user-tour/react-user-tour.d.ts]: "A hex value" seems to be string or number, not `any`

**Comment by ccancellieri** [comment]
Thanks for the super detailed review, I'll try to use an automatic formatting tool next time

**Comment by zhengbli** [comment]
👍

**Pull request merged**